### PR TITLE
Twit input

### PR DESCRIPTION
**Pull request by dranilew**
Currently, when the user inputs something, it does not redirect to any other parts of the website. However, for the time being, this will do for the start of the prototype.

### CODE REVIEW

File: src/main/java/com/google/sps/servlets/KeywordServlet.java
Comment (nssafir) [inline src/main/java/com/google/sps/servlets/KeywordServlet.java:20]: Should we use a list data structure for now if we're starting with one tweet, or should we stick with just a single string?

File: src/main/java/com/google/sps/servlets/KeywordServlet.java
Comment (nssafir) [inline src/main/java/com/google/sps/servlets/KeywordServlet.java]: Should there be some type of check that the user input is valid?

File: src/main/java/com/google/sps/servlets/KeywordServlet.java
Comment (nssafir) [inline src/main/java/com/google/sps/servlets/KeywordServlet.java]: [Nit] delete newlines

File: src/main/java/com/google/sps/servlets/KeywordServlet.java
Comment (nssafir) [inline src/main/java/com/google/sps/servlets/KeywordServlet.java:57]: [Nit] add new line

File: src/main/webapp/scripts/load.js
Comment (nssafir) [inline src/main/webapp/scripts/load.js]: [Nit] delete new line

File: src/main/webapp/scripts/load.js
Comment (nssafir) [inline src/main/webapp/scripts/load.js]: [Nit] add new line

File: src/main/webapp/index.html
Comment (nssafir) [inline src/main/webapp/index.html]: You may want to add an external css style sheet.

File: src/main/webapp/scripts/load.js
Comment (dranilew) [reply src/main/webapp/scripts/load.js]: I think the new line before the comment block improves readability.

File: src/main/webapp/index.html
Comment (dranilew) [reply src/main/webapp/index.html]: I will put this in index.css instead.

File: src/main/java/com/google/sps/servlets/KeywordServlet.java
Comment (dranilew) [reply src/main/java/com/google/sps/servlets/KeywordServlet.java:20]: I think for the purposes of the prototype, this would do. Perhaps I'll make it so that the textarea in which you input the tweet is pre-filled with the "hard-coded" tweet.

**Comment by daperlmut** [comment]
This doesn't look like it was approved by anyone yet.  Please try to wait until there is at least one approval before requesting me/Sean to review.

File: src/main/webapp/index.html
Comment (juanuribe28) [inline src/main/webapp/index.html]: I think this from should probably go in a different page. The user should log in or input the tweet in one page, and then get redirected to index.html where the keywords, news and actions are shown.

File: src/main/webapp/styles/index.css
Comment (juanuribe28) [inline src/main/webapp/styles/index.css]: [Optional] Use vh and vw units instead of px. This way the size of the element adapts to the size of the screen.

File: src/main/webapp/scripts/load.js
Comment (juanuribe28) [inline src/main/webapp/scripts/load.js]: I don't think this function is needed as toggle.js manages the sections' loading.

File: src/main/java/com/google/sps/servlets/KeywordServlet.java
Comment (nssafir) [reply src/main/java/com/google/sps/servlets/KeywordServlet.java:20]: Actually just looked at the new design doc comments and I think I misunderstood, the way you have it is good.

File: src/main/webapp/index.html
Comment (dranilew) [reply src/main/webapp/index.html]: I will move it to the keywords page for now.

File: src/main/webapp/index.html
Comment (juanuribe28) [inline src/main/webapp/index.html]: Why did you remove this div?

Comment (dranilew) [reply src/main/webapp/index.html]: Oh shoot, I meant to remove the forms, not the whole thing. Mb, I'll re-add it.

**Review by juanuribe28 — APPROVED** [review]

**Review by juanuribe28 — APPROVED** [review]

File: src/main/java/com/google/sps/servlets/KeywordServlet.java
Comment (daperlmut) [inline src/main/java/com/google/sps/servlets/KeywordServlet.java]: [Nit] I think there is a minor grammar issue in this sentence which you can fix by just removing the words "from those"

File: src/main/java/com/google/sps/servlets/KeywordServlet.java
Comment (daperlmut) [inline src/main/java/com/google/sps/servlets/KeywordServlet.java]: [Nit] Missing a '.'

File: src/main/java/com/google/sps/servlets/KeywordServlet.java
Comment (daperlmut) [inline src/main/java/com/google/sps/servlets/KeywordServlet.java]: I think this would be much easier to read using String.format
See https://www.geeksforgeeks.org/java-string-format-examples/

Also, you can turn the builder into an actual StringBuilder object
See https://docs.oracle.com/javase/7/docs/api/java/lang/StringBuilder.html

File: src/main/java/com/google/sps/servlets/KeywordServlet.java
Comment (daperlmut) [inline src/main/java/com/google/sps/servlets/KeywordServlet.java]: Just to confirm, this is Java 7 right?  I was thinking about suggestion the join function but I think it only works on lists in Java 8
https://reversecoding.net/java-8-convert-list-string-comma/

File: src/main/java/com/google/sps/servlets/KeywordServlet.java
Comment (daperlmut) [inline src/main/java/com/google/sps/servlets/KeywordServlet.java:50]: [Nit] Remove new line between braces

**Review by daperlmut — CHANGES_REQUESTED** [review]

File: src/main/java/com/google/sps/servlets/KeywordServlet.java
Comment (dranilew) [reply src/main/java/com/google/sps/servlets/KeywordServlet.java]: Yea, but I feel like using the join() function makes this less intuitive/readable to the reader, since it'd look like
`String.join("},{\"term\": ", toJson);` which I feel like is harder to read than simply String.format(//stuffs here).

Comment (daperlmut) [reply src/main/java/com/google/sps/servlets/KeywordServlet.java]: Assuming the join syntax works on lists you can just join on ",".  In Java 8 I would do this

```
List<String> terms = new ArrayList<string>();
for (String term : toJson) {
  terms.add(String.format("{\"term\": \"%s\"}", term));
}
String keywordPrefix = "{\"keywords\": [";
return String.format("{\"keywords\": [%s]}", terms.join(","))
```

You might also want to factor a function that, given a key and value return "{\"key\": value}" to make it easier to build JSON objects.  Alternatively, you can use a library that builds JSON for you.  A quick google search turns up 
this: https://docs.oracle.com/javaee/7/api/javax/json/JsonObjectBuilder.html
and this: https://github.com/stleary/JSON-java

**Review by daperlmut — CHANGES_REQUESTED** [review]

File: src/main/java/com/google/sps/servlets/KeywordServlet.java
Comment (seangilpin) [inline src/main/java/com/google/sps/servlets/KeywordServlet.java]: I don't see this as being inherently tied to mustache. It's creating a json encoding of the terms?

File: src/main/webapp/json/keywords.json
Comment (seangilpin) [inline src/main/webapp/json/keywords.json:3]: is this meant to be keywords instead of keyterms?

File: src/main/java/com/google/sps/servlets/KeywordServlet.java
Comment (daperlmut) [inline src/main/java/com/google/sps/servlets/KeywordServlet.java]: [Nit] I include the keywordPrefix variable in my suggestion but then realized it wasn't necessary because it is only used one.  I would just hard code it here.

Comment (daperlmut) [reply src/main/java/com/google/sps/servlets/KeywordServlet.java]: Also, I think you can use your helper function in the return here now.

`return keyValueJson("keywords", String.format("[%s]", String.join(",", terms))`

I'm still partial to using an actual JSON library for this, but you can do that in a future PR if you want.

**Review by daperlmut — CHANGES_REQUESTED** [review]

File: src/main/java/com/google/sps/servlets/KeywordServlet.java
Comment (dranilew) [reply src/main/java/com/google/sps/servlets/KeywordServlet.java]: Yea, I realized that and removed the prefix variable, but it's not pushed yet (Might be something you see in the Saliency branch as Nathan mentioned something similar to this there.

As for the JSON library, I will probably leave that for a future PR.

Comment (daperlmut) [reply src/main/java/com/google/sps/servlets/KeywordServlet.java]: Ok, I'll approve this PR as soon the changes to remove keywordPrefix variable are committed (I'm assuming that change is going to be made to this PR)

**Review by daperlmut — CHANGES_REQUESTED** [review]

**Review by daperlmut — APPROVED** [review]

**Pull request merged**
